Sources/PostgresNIO/New/PSQLConnection.swift

```diff
@@ -137,7 +139,15 @@ final class PSQLConnection {
             promise: promise)
         
         self.channel.write(PSQLTask.extendedQuery(context), promise: nil)
-        return promise.futureResult
+        return promise.futureResult.always { result in
+            switch result {
+            case .failure(let error):
+                logger.error("Query failed", metadata: [.error: "\(error)"])
+            case .success:
+                // success is logged in PSQLQuery
+                break
+            }
+        }
```

Comment:
The success case doesn't do anything here, use `whenFailure(_:)` instead.
```suggestion
        // success is logged in PSQLQuery
        return promise.futureResult.whenFailure { error in
            logger.error("Query failed", metadata: [.error: "\(error)"])
        }
```